src/app/core/services/data.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+
+import { Videos } from '@models/video.model';
+import { VimeoResponse } from '@models/vimeo.model';
+import { YouTubeResponse } from '@models/youtube.model';
+import { Platform } from '@shared/platform.model';
+import { LocalStorageService } from './local-storage.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DataService {
+    public userVideosList: Videos = [];
+    public subject = new Subject<Videos>();
+    
+    private userVideosCounter = this.userVideosList.length;
+    private love = false;
+    
+    constructor(private localStorageService: LocalStorageService) {}
+
+    public addYouTubeVideo(resp: YouTubeResponse): void {
+        this.userVideosList.push(
+            {
+                platform: Platform.youtube,
+                id: ++this.userVideosCounter,
+                favorite: false,
+                date: new Date(),
+                videoId: resp.items[0].id,
+                title: resp.items[0].snippet.title,
+                img: resp.items[0].snippet.thumbnails.default.url,
+                viewCount: resp.items[0].statistics.viewCount
+            }
+        );
+        this.subject.next(this.userVideosList);
+        this.localStorageService.saveToLocalStorage(this.userVideosList);
+    }
+    
+    public addVimeoVideo(resp: VimeoResponse): void {
+        this.userVideosList.push(
+            {
+                platform: Platform.vimeo,
+                id: ++this.userVideosCounter,
+                favorite: false,
+                date: new Date(),
+                videoId: resp.video_id.toString(),
+                title: resp.title,
+                img: resp.thumbnail_url,
+                viewCount: "Brak danych"
```

Comment:
tutaj trochę magic string no ale gdybyśmy mieli tłumaczenia to pewnie wartość z tłumaczenia byśmy tutaj przypisali więc ostatecznie może tak zostać